### PR TITLE
Media: Add filter to disable potentially expensive tax query

### LINKED ISSUE #9853: Media: tax query affects WP media library performance
## Bug Description

> The Web Stories plugin is causing a performance issue with the WordPress Media Library.  The issue is a tax query that's being added in the function `get_exclude_tax_query` in file `plugins/web-stories/includes/Media/Media_Source_Taxonomy.php`
> 
> This tax query is causing the underlying SQL query to be too slow and causing long delays with working in the media library.  This is most noticeable in the preprod environment due to lower database resources, but is slowing down the media library in production as well. 
> 
> We would like to have the plugin add a filter that would allow us to disable the addition of this tax query.  We tested changing the plugin directly to remove this tax query and that appears to resolve the issue.  There are three functions where we had to make code changes for this: `filter_ajax_query_attachments_args`, `filter_generated_media_attachments` and `filter_rest_generated_media_attachments`
> 
> We would need this filter to apply to all of these contexts.  

## Expected Behaviour

Media library performance is good.

## Steps to Reproduce

See above.

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add a filter to disable attachment filtering. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9853

### CODE REVIEW

File: includes/Media/Media_Source_Taxonomy.php [hunk 1]
```diff
@@ -213,6 +213,19 @@ public function update_callback_media_source( $value, $object ) {
...
 	private function get_exclude_tax_query( array $args ): array {
+		/**
+		 * Filter whether the query should be filtered.
+		 *
+		 * @since 1.16.0
+		 *
+		 * @param bool  $enabled Whether the taxonomy check should be applied.
+		 * @param array $args    Existing WP_Query args.
+		 */
+		$enabled = apply_filters( 'web_stories_enable_exclude_query', true, $args );
```
Comment (swissspidy) [inline includes/Media/Media_Source_Taxonomy.php]: WDYT about a filter name like `web_stories_hide_auto_generated_attachments` that explains a bit more what this actually does?

File: includes/Media/Media_Source_Taxonomy.php [hunk 2]
```diff
@@ -213,6 +213,19 @@ public function update_callback_media_source( $value, $object ) {
 	 * @return array  Tax query arg.
 	 */
 	private function get_exclude_tax_query( array $args ): array {
+		/**
+		 * Filter whether the query should be filtered.
```
Comment (swissspidy) [inline includes/Media/Media_Source_Taxonomy.php]:
```suggestion
		 * Filter whether generated attachments should be hidden in the media library.
```

**Review by swissspidy — APPROVED** [review]
Just a question about naming

File: includes/Media/Media_Source_Taxonomy.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Media/Media_Source_Taxonomy.php]: I like it. https://github.com/google/web-stories-wp/pull/9991/commits/1379c516123f70eb5b83dee7316f8f84a27de3c9

**Comment by swissspidy** [comment]
Haven't heard back yet whether this filter suffices, but seems reasonable to add nonetheless.

**Pull request merged**